cetmix_tower_server/models/cx_tower_server.py

```diff
@@ -1550,68 +1553,70 @@ def _run_python_code(
+                - 'p' = with password (returns list),
+                - 'n' = no password (returns string),
+                - False/None = no sudo.
+            **kwargs:
+                no_split_for_sudo (bool): If True, disables splitting by '&&'.
 
         Returns:
-            command (list|str): command splitted into separate commands
-                (for sudo with password mode ('p')) or composed command
-                from its parts for sudo without password mode ('n'))
-
+            str or list[str]: SSH-ready command(s) to be executed.
         """
-        # Prepare command for sudo if needed
-        if sudo:
-            # Add location
-            sudo_prefix = "sudo -S -p ''"
+        no_split = kwargs.get("no_split_for_sudo", False)
+        sudo_prefix = "sudo -S -p ''"
 
-            # Detect command separator
-            separator = "&&"
-            if separator in command_code:
-                result = (
-                    command_code.replace("\\", "").replace("\n", "").split(separator)
-                )
-
-                # Sudo with password expects a list of commands
-                result = [f"{sudo_prefix} {cmd.strip()}" for cmd in result]
+        # Normalize the command: remove newlines and trim spaces, preserve backslashes
+        command_code = " ".join(
+            line.strip() for line in command_code.splitlines() if line.strip()
+        )
 
-                # Merge back into a single command is sudo is without password
-                if sudo == "n":
-                    result = f" {separator} ".join(result)
-            else:
-                # Single command only
+        if sudo:
+            # Case 1: Do not split - wrap the whole command with sudo
+            if no_split or "&&" not in command_code:
                 result = f"{sudo_prefix} {command_code}"
 
-                if sudo == "p":
-                    # Sudo with password expects a list of commands
-                    result = [result]
+            # Case 2: Split command by '&&' and wrap each segment
+            else:
+                parts = [
+                    f"{sudo_prefix} {part.strip()}" for part in command_code.split("&&")
+                ]
+                result = parts if sudo == "p" else " && ".join(parts)
 
+            # Make sure `sudo='p'` always returns a list
+            if sudo == "p" and isinstance(result, str) and not no_split:
+                result = [result]
         else:
-            # Command without sudo is always run as is
             result = command_code
-        # Add path change command
+
+        # If path is given, prepend it to the command(s)
         if path:
-            # Add sudo prefix if needed
             cd_command = f"cd {path}"
-
             if isinstance(result, list):
-                result = [cd_command] + result
+                result = (
+                    f"{cd_command} && {result[0]}"
+                    if no_split
```

Comment:
Question: what's the reason behind this action?

Reply:
Ok, but we don't write code to pass the tests. We write tests to check the code.
Have you checked what happens with these values next?